ddp-workspace/projects/ddp-sdk/src/lib/services/fileUpload.service.ts

```diff
@@ -21,23 +21,24 @@ export class FileUploadService extends UserServiceAgent<any> {
         super(session, configuration, http, logger, null);
     }
 
-    getUploadUrl(studyGuid: string, activityGuid: string, questionStableId: string, file: File): Observable<FileUploadResponse> {
+    getUploadUrl(studyGuid: string, activityGuid: string, questionStableId: string, files: File[]): Observable<FileUploadResponse[]> {
         const path = `/studies/${studyGuid}/activities/${activityGuid}/uploads`;
 
-        const requestBody: FileUploadBody = {
-            questionStableId,
-            fileName: file.name,
-            fileSize: file.size,
-            mimeType: file.type
-        };
-
-        return this.postObservable(path, requestBody, {}, true).pipe(
-            catchError(error => {
-                this.logger.logDebug('getUploadUrl error', error);
-                return throwError(error.error);
+        const acceptedFiles: Observable<FileUploadResponse | null>[] = files.map<FileUploadBody>(file =>  ({
```

Comment:
We usually use **_name$_** for Observable.
`acceptedFiles` => `acceptedFiles$` would be nice.

Reply:
I used to as well, don't know why I missed it here :(. Done